### PR TITLE
contrib: std::exception can't be catched correctly with libc++

### PR DESCRIPTION
**Pull request by durd07**
Signed-off-by: Felix Du <felix.du@nokia-sbell.com>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/21497
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by daixiang0** [comment]
The PR title confuse me: is it only related to contrib part? And what is the problem now, remove it works for all cases?

**Comment by durd07** [comment]
> The PR title confuse me: is it only related to contrib part? And what is the problem now, remove it works for all cases?

is it only related to contrib part?
--> yes
And what is the problem now,
--> coredump occured if there is some code throw exceptions
remove it works for all cases
--> I don't know how to cover all cases and what does the all cases mean? But all checks have passed.

Details please refer to this issue: https://github.com/envoyproxy/envoy/issues/21497

File: bazel/foreign_cc/BUILD [hunk 1]
```diff
@@ -150,7 +150,6 @@ configure_make(
         "ICU_DATA_FILTER_FILE": "$(execpath //bazel/foreign_cc:icu_data_filter.json)",
     },
     lib_source = "@com_github_unicode_org_icu//:all",
-    linkopts = ["-lstdc++"],
```
Comment (KBaichoo) [inline bazel/foreign_cc/BUILD:153]: wonder if this should have this in the CXXFLAGs as in:

`CXXFLAGS": "-lstdc++` used for ragel. That seems to have worked in this case: https://github.com/envoyproxy/envoy/pull/19473#discussion_r790427535 using the `ENVOY_STDLIB` override.

Will defer to the expert (Lizan) in this area.

**Comment by lizan** [comment]
Is there anyway to test/verify this so we don't get regression?

**Comment by durd07** [comment]
@diceride Could you please help to verify whether this PR will block your commit https://github.com/envoyproxy/envoy/pull/20143?

@lizan Even though I don't know how to verify this PR will affect features involved from PR https://github.com/envoyproxy/envoy/pull/20143(need owner's help to verify), There is obvious error that both link libc++ and libstdc++ at the same time. Actually PR20143 has already block all contrib extensions. 

At least compile is OK and all CI pipelines are passed.

**Comment by durd07** [comment]
> What do you mean by "block all contrib extensions"? Their tests passes in main branch and here.

Yes, As I described in the issue https://github.com/envoyproxy/envoy/issues/21497, **All try/catch** in the contrib image doesn't work.

> Their tests passes in main branch and here.
That's right, but there is no cases to touch the `try/catch` exceptions. 

So I think the solution is merge my PR ASAP or revert the origianl PR https://github.com/envoyproxy/envoy/pull/20143

**Comment by lizan** [comment]
@durd07 Can you port this test for contrib? https://github.com/envoyproxy/envoy/blob/main/test/exe/envoy_static_test.sh

**Review by lizan — APPROVED** [review]

**Comment by durd07** [comment]
@KBaichoo @lizan Could you help to merge this PR?

**Comment by durd07** [comment]
> @durd07 Can you port this test for contrib? https://github.com/envoyproxy/envoy/blob/main/test/exe/envoy_static_test.sh

I can try to do it later, but seems there is no such framework for contrib, we can discuss how to port some test of the official binary into contrib. There is less of quality check of contrib.

File: bazel/foreign_cc/BUILD (hunk 1 above)
Comment (zhxie) [reply bazel/foreign_cc/BUILD:153]: I think if the `configure_make` does not produce any binaries like ragel, then it is safe to remove `-lstdc++`.

Comment (durd07) [reply bazel/foreign_cc/BUILD:153]: @zhxie It is true that there won't be any other binaries generated besides envoy-static, so it is safe to remove `-lstdc++`.

**Pull request merged**